ServerCore/Pages/Events/Map.cshtml.cs

```diff
@@ -21,17 +21,28 @@ public class MapModel : EventSpecificPageModel
 
         public StateStats[,] StateMap { get; private set; }
 
+        public string Groups { get; set; }
+
         public MapModel(PuzzleServerContext serverContext,
                         UserManager<IdentityUser> userManager)
             : base(serverContext, userManager) { }
 
-        public async Task<IActionResult> OnGetAsync(int? refresh)
+        public async Task<IActionResult> OnGetAsync(int? refresh, string? groups)
         {
             if (refresh != null)
             {
                 Refresh = refresh;
             }
 
+            Groups = groups ?? "Meta|*";
+
+            Dictionary<string, int> groupSortIndex = new Dictionary<string, int>();
+            string[] groupList = Groups.Split("|");
+            for (int i = 0; i < groupList.Length; i++)
+            {
+                groupSortIndex[groupList[i]] = i;
```

Comment:
Consider trimming the strings so there isn't whitespace confusion. Maybe also case/space insensitive comparison?

Reply:
Good suggestions, done!